src/Psalm/Internal/Analyzer/Statements/Expression/BooleanNotAnalyzer.php

```diff
@@ -33,12 +33,11 @@ public static function analyze(
         $stmt_type = Type::getBool();
         if ($expr_type) {
             if ($expr_type->isAlwaysTruthy()) {
-                $stmt_type = Type::getFalse();
+                $stmt_type = Type::getFalse($expr_type->from_docblock);
             } elseif ($expr_type->isAlwaysFalsy()) {
-                $stmt_type = Type::getTrue();
+                $stmt_type = Type::getTrue($expr_type->from_docblock);
             }
 
-            $stmt_type->from_docblock = $expr_type->from_docblock;
             $stmt_type->parent_nodes = $expr_type->parent_nodes;
```

Comment:
Did I miss something or this should be flagged as an invalid access?

Reply:
Not yet, as the Union class isn't @psalm-immutable yet.